src/aind_metadata_mapper/neuropixels/rig.py

```diff
@@ -0,0 +1,197 @@
+                    mvr_name
+                )
+        return extracted
+
+    def _extract_sync(self, content: str) -> SyncContext:
+        """Extracts DAQ-related information from MPE sync config.
+        """
+        config = yaml.safe_load(content)
+        return (
+            config["device"],
+            config["freq"],
+            list(config["line_labels"].items()),
+        )
+
+    def _transform(self, extracted_source: RigContext) -> rig.Rig:
+        """Transforms extracted rig context into aind-data-schema rig.Rig
+        instance.
+        """
+        partial, mvr_cameras, sync_context = extracted_source
+
+        # search for partial sync daq
+        sync_device_name, sync_sample_rate, sync_channels = sync_context
+        sync_daq_name = "Sync"
+        for idx, partial_daq in enumerate(partial["daqs"]):
+            if partial_daq["name"] == sync_daq_name:
+                # remove from daqs for later spread operation
+                partial_sync_daq = partial["daqs"].pop(idx)
+                break
+        else:
+            raise NeuropixelsRigException(
+                "Sync daq not found in partial rig. expected=%s" %
+                sync_daq_name
+            )
+
+        sync_daq = {
+            **partial_sync_daq,
+            "channels": [
+                {
+                    "channel_name": name,
+                    "channel_type": "Digital Input",
+                    "device_name": sync_device_name,
+                    "event_based_sampling": False,
+                    "channel_index": line,
+                    "sample_rate": sync_sample_rate,
+                    "sample_rate_unit": "hertz"
+                }
+                for line, name in sync_channels
+            ],
+        }
+
+        camera_assemblies = []
+        for partial_camera_assembly in partial["cameras"]:
+            name = partial_camera_assembly["camera_assembly_name"]
+
+            found = list(filter(
+                lambda camera: camera[0] == name,
+                mvr_cameras,
+            ))
+            if len(found) < 1:
+                raise NeuropixelsRigException(
+                    "Camera assembly not found in partial rig. expected=%s"
+                    % name
+                )
+
+            assembly_name, serial_number, height, width, frame_rate = found[0]
+            camera_assemblies.append({
+                **partial_camera_assembly,
+                "camera": {
+                    **partial_camera_assembly["camera"],
+                    "name": assembly_name,
+                    "serial_number": serial_number,
+                    "pixel_height": height,
+                    "pixel_width": width,
+                    "size_unit": "pixel",
+                    "max_frame_rate": frame_rate,
+                }
+            })
+
+        return rig.Rig.parse_obj({
+            **partial,
+            "modification_date": datetime.date.today(),
```

Comment:
add `modification_date` to the arg list. you can mention in the defaults that `None` means `today()`.